windows/src/engine/keyman32/CoreEnvironment.cpp

```diff
@@ -0,0 +1,32 @@
+#include "pch.h"
+
+BOOL SetupCoreEnvironment(km_kbp_option_item **core_environment) {
+  km_kbp_option_item *items = new km_kbp_option_item[5];
+
+  items[0].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[0].key = KM_KBP_KMX_ENV_BASELAYOUT;
+  items[0].value = reinterpret_cast<km_kbp_cp*>(Globals::get_BaseKeyboardName());
+
+  items[1].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[1].key = KM_KBP_KMX_ENV_BASELAYOUTALT;
+  items[1].value = reinterpret_cast<km_kbp_cp*>(Globals::get_BaseKeyboardNameAlt());
+
+  items[2].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[2].key = KM_KBP_KMX_ENV_SIMULATEALTGR;
+  items[2].value = Globals::get_SimulateAltGr() ? u"1" : u"0";
+
+  items[3].scope = KM_KBP_OPT_ENVIRONMENT;
+  items[3].key = KM_KBP_KMX_ENV_BASELAYOUTGIVESCTRLRALTFORRALT;
+  items[3].value = KeyboardGivesCtrlRAltForRAlt() ? u"1" : u"0";
+
+  items[4] = KM_KBP_OPTIONS_END;
```

Comment:
On Linux we also set `KM_KBP_KMX_ENV_PLATFORM` - I assume we don't need that on Windows?

Reply:
ok, I see. It uses the default which happens to be "windows hardware desktop native". Seems to me it would be cleaner and clearer to explicitly set it...

Reply:
I agree, we should add that as well. I might do it as a separate PR?